### DISCUSSION TITLE
Is it possible to set different options for models under the same endpoint, like icon and display name?

### DISCUSSION THREAD

**Discussion by erickrf** [danny-avila/LibreChat#4108] [Q&A] [answered]
I am working with a custom endpoint serving a few very different models, and it would be great to be able to set icons and display names for each of them separately. I cannot see how this is supported in the [custom endpoint docs](https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/custom_endpoint).

Since most of the models don't have their names automatically recognized by LibreChat, the chats with them only display "AI" (or whatever default name I set), which doesn't allow me to tell later on which one was used when.

I can imagine the workaround of copying the endpoint and change its configurations for each model. That has the inconvenient that the users would have to reenter their API key for all models.

**Comment by danny-avila** [comment, marked as answer]
See this: https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/model_specs

**Reply by erickrf** [reply]
That works but now I realize there's no way that the user can input their API key, even when modelSpecs has `enforce: false` and `prioritize: false`. Is there still a way to allow user-given keys?

**Comment by erickrf** [comment]
Thanks! That's exactly what I needed.

I had briefly read that page before, but some things were unclear to me and I didn't realize it was meant for this use case. I'd suggest explaining a bit more what you can do with this config at the top of the page, and that's something you do _in addition to_ setting the endpoints.

Most of the fields are well documented and easy to understand, but the two at the very beginning not so much. It's not clear which configurations are referred to by `enforce` and `prioritize`.
